pkg/registry/service/rest.go

```diff
@@ -133,6 +135,35 @@ func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		}
 	}
 
+	if shouldCheckOrAssignHealthCheckNodePort(service) {
+		var healthCheckNodePort int
```

Comment:
please add a unittest to service/rest_test where you actually invoke this Create method with various services and verify the output (with node port, with annotation set to garbage, clusterIP etc)